### PR TITLE
[RCM] Fix rc config deletion

### PR DESCRIPTION
**Pull request by coignetp**
### What does this PR do?

When the main RC client in the agent received an update from the backend to delete a config, there were 2 bugs:
* If `config delete` was the only update, there were no state update, we just ignored it
* Else, if `config delete` was the only update of a product, the state was updated, but the RC client listeners were not called with the update

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/remoteconfig/state/repository.go [hunk 1]
```diff
@@ -256,8 +262,8 @@ func (r *Repository) Update(update Update) ([]string, error) {
 	}
 
 	changedProducts := make([]string, 0)
-	for product, configs := range result.changed {
-		if len(configs) > 0 {
+	for product, updated := range result.productsUpdated {
+		if updated {
```
Comment (BaptisteFoy) [inline pkg/remoteconfig/state/repository.go:266]: **🥜 Nitpick:** Is that necessary? If a key is in the map, it means the product has been updated

File: pkg/config/remote/client.go [hunk 2]
```diff
@@ -363,9 +363,10 @@ func (c *Client) newUpdateRequest() (*pbgo.ClientGetConfigsRequest, error) {
 	pbConfigState := make([]*pbgo.ConfigState, 0, len(state.Configs))
 	for _, f := range state.Configs {
 		pbConfigState = append(pbConfigState, &pbgo.ConfigState{
-			Id:      f.ID,
-			Version: f.Version,
-			Product: f.Product,
+			Id:         f.ID,
+			Version:    f.Version,
+			Product:    f.Product,
+			ApplyState: uint64(f.ApplyStatus.State),
```
Comment (BaptisteFoy) [inline pkg/config/remote/client.go:369]: **❓ Question:** ‏It's good that we report the configuration status, but is it required to fix the bugs in this PR?

**Review by BaptisteFoy — APPROVED** [review]

File: pkg/config/remote/client.go [hunk 3]
```diff
@@ -215,6 +215,11 @@ func (c *Client) Close() {
 	c.close()
 }
 
+// UpdateApplyStatus updates the config's metadata to reflect its applied status
+func (c *Client) UpdateApplyStatus(cfgPath string, status state.ApplyStatus) {
+	c.state.UpdateApplyStatus(cfgPath, status)
+}
+
```
[the remark this answers is not in the document]
Comment (coignetp) [reply pkg/config/remote/client.go]: Forgot to remove it from my changes indeed

File: pkg/remoteconfig/state/repository.go (hunk 1 above)
Comment (coignetp) [reply pkg/remoteconfig/state/repository.go:266]: It's to be future-proof, and it is not really costly

**Review by ameske — APPROVED** [review]

**Pull request merged**